### PR TITLE
Add led_array modifiers

### LINKED ISSUE #5: Add LED modifiers
A way to gradually layer modifiers ontop of a led array to step-wise work towards a unique look on the leds.
A starting selection could be:
* BaseColourSelect(colour, brightness)
* ColourVariation(variation_amount int, tick)
* BreathAnimation(maxDiff, minDiff)
* ProgressAnimation(progress) <- ]0, 100[

related to #4

### PR DESCRIPTION
**Pull request by LeonEcke**
Adds modifiers that take an array of HSV colours and modifies them in different ways.
These modifiers can be stacked to create different looking outcomes from the same functions.

Completes #5.

### CODE REVIEW

File: firmware/src/ledModifier.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	for (int i = 0; i < nr_of_leds; i++) {
+		led_array[i] = colour;
+	}
+}
+
+void changeHue(HSV &led, int amount) {
+	led.hue = ((led.hue + amount) + 255) % 255;	 // looping
+}
+
+void changeSaturation(HSV &led, int amount) {
```
Comment (tobyjw) [inline firmware/src/ledModifier.cpp:30]: You can refactor this to a ternary statement.
`led.saturation = (led.saturation + amount < 0) ? 0 : 255`

https://www.w3schools.com/cpp/cpp_conditions_shorthand.asp

File: firmware/src/ledModifier.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	int capped_amount = led.saturation + amount;
+	if (capped_amount < 0)
+		capped_amount = 0;
+	else if (capped_amount > 255)
+		capped_amount = 255;
+	led.saturation = capped_amount;
+}
+
+void changeValue(HSV &led, int amount) {
```
Comment (tobyjw) [inline firmware/src/ledModifier.cpp:39]: See above on ternary statements

File: firmware/src/ledModifier.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+void setColour(HSV colour, uint8_t bitmask) {
+	for (int i = 0; i < nr_of_leds; i++) {
+		led_array[i] = colour;
+	}
+}
+
+void changeHue(HSV &led, int amount) {
+	led.hue = ((led.hue + amount) + 255) % 255;	 // looping
```
Comment (tobyjw) [inline firmware/src/ledModifier.cpp]: I notice `255` being used in a few places. This is what we call a "magic number" which can (or should) be stored as a const value with a useful name to make its purpose clearer to the reader.

https://www.inf.unibz.it/~calvanese/teaching/04-05-ip/lecture-notes/uni04/node15.html

File: firmware/src/ledModifier.cpp (hunk 1 above)
Comment (LeonEcke) [reply firmware/src/ledModifier.cpp:30]: Wouldnt that set it to either 0 or 255, with no inbetween? Theres cases where `0 < led.saturation + amount < 255`.

File: firmware/src/ledModifier.cpp (hunk 3 above)
Comment (LeonEcke) [reply firmware/src/ledModifier.cpp]: Good point. Implemented.

**Pull request merged**